general/src/macros.rs

```diff
@@ -39,23 +39,23 @@ macro_rules! numeric_meta_structure {
     ($type:ident, $meta_type:ident, $expectation:expr, $process_func:ident) => {
         impl crate::processor::FromValue for $type {
             fn from_value(value: Annotated<Value>) -> Annotated<Self> {
-                match value {
-                    Annotated(Some(Value::U64(value)), meta) => {
-                        Annotated(Some(value as $type), meta)
-                    }
-                    Annotated(Some(Value::I64(value)), meta) => {
-                        Annotated(Some(value as $type), meta)
-                    }
-                    Annotated(Some(Value::F64(value)), meta) => {
-                        Annotated(Some(value as $type), meta)
-                    }
-                    Annotated(Some(Value::Null), meta) => Annotated(None, meta),
-                    Annotated(None, meta) => Annotated(None, meta),
-                    Annotated(Some(value), mut meta) => {
-                        meta.add_unexpected_value_error($expectation, value);
-                        Annotated(None, meta)
+                value.and_then(|value| {
+                    let number: Option<$type> = match value {
```

Comment:
Just wondering, is this type declaration required?

Reply:
nah, it was a leftover from when I confused the compiler into confusing me